Validate post-quantum key types when choosing encryption keys for lease sets.

Index: libi2pd/LeaseSet.cpp
--- libi2pd/LeaseSet.cpp.orig
+++ libi2pd/LeaseSet.cpp
@@ -9,6 +9,7 @@
 #include <string.h>
 #include "I2PEndian.h"
 #include "Crypto.h"
+#include "PostQuantum.h"
 #include "Log.h"
 #include "Tag.h"
 #include "Timestamp.h"
@@ -420,7 +421,10 @@ namespace data
 				i2p::crypto::GetCryptoPublicKeyLen (keyType) == encryptionKeyLen) // key length mismatch or unknown key type
 			{
 				// we pick max key type if preferred not found
-#if !OPENSSL_PQ
+#if OPENSSL_PQ
+				if (keyType <= i2p::data::CRYPTO_KEY_TYPE_ECIES_X25519_AEAD ||
+					i2p::crypto::IsMLKEMSupported (keyType))
+#else
 				if (keyType <= i2p::data::CRYPTO_KEY_TYPE_ECIES_X25519_AEAD) // skip PQ keys if not supported
 #endif
 				{
